### DISCUSSION TITLE
cannot display battery or time

### DISCUSSION THREAD

**Discussion by AntouanK** [starship/starship#1854] [Q&A] [answered]
Hi there.

Trying to understand how to config starship.

I have this config right now
```
❯ cat .config/starship.toml 

[gcloud]
disabled = true

[battery]
disabled = false
full_symbol = "🔋"
charging_symbol = "⚡️"
discharging_symbol = "💀"
unknown = "??"
empty = "__"

[battery.display]
threshold = 10
style = "bold red"

[time]
disabled = false
format = '🕙[\[ $time \]]($style) '
time_format = "%T"
utc_time_offset = "-5"
time_range = "10:00:00-14:00:00"
```

It disables gcloud, so starship does read the config.
I tried removing options from battery and time, and still no change at all.

on the home directory I get 
```
~ 
❯ RUST_LOG=starship starship prompt

%{%}~%{%} 
%{%}❯%{%} %                                                              
~ 
❯
```

any idea why I cannot see the battery or time?

thank you.

**Comment by mickimnet** [comment]
Hello @AntouanK,

what happens if you do the following:
- **for the battery:** change `threshold = 10` to `threshold = 100`
- **for the time:** remove/comment the line with `time_range = "10:00:00-14:00:00"`

Take care
Mick

**Reply by AntouanK** [reply]
huh, I guess I didn't understand the config.

Doing those two changes, does show the time.
```
~ 
🕙[ 02:39:49 ] ❯ RUST_LOG=starship starship prompt

%{%}~%{%} 
🕙%{%}[ %{%}%{%}02:40:24%{%}%{%} ]%{%} %{%}❯%{%} %                       
~ 
🕙[ 02:40:24 ] ❯ 
```

( icon alignment is a bit weird, but that's a separate issue I suppose )
![Screenshot from 2020-11-05 08-41-08](https://user-images.githubusercontent.com/4569111/98211692-b468c600-1f42-11eb-840f-00be89153a8e.png)

**Comment by mickimnet** [comment, marked as answer]
The `time_range` option does define during which time frame you would like to have the time info displayed. Which is quite nice if you only want to have the time section shown during a specific time frame (e. g. from 10 am to 2 pm in your config from above).

The `battery.display` section with the `threshold` option defines at what percentage of the battery, the section should be shown. You can have more `battery.display` sections with different values. For further information see the [configuration documentation](https://starship.rs/config/#battery).

**Reply by AntouanK** [reply]
got it, thank you.

2 questions that I cannot figure out from the docs.
- is it `[[battery.display]]` or `[battery.display]`?
- I try threshold 1 or 100, the battery status never appears. Should it appear? ( I have 32% of battery right now )

**Reply by mickimnet** [reply]
It is `[[battery.display]]`. Try out the following:

```toml
[battery]
format = "[$symbol$percentage]($style) "
[[battery.display]]
threshold = 50
style = "bold red"
[[battery.display]]
threshold = 100
style = "white"
```

And, do you have a `format = ` section in the top of your `starship.toml`? If so, could you please post it?

Thanks & take care
Mick

**Reply by AntouanK** [reply]
that worked, thank you so much.

no, I didn't have any `format = ` at the top

**Reply by mickimnet** [reply]
Great that it worked out!

**Comment by 52LONG** [comment]
Thank you very much for sharing the problem, and thank you very much for the brother who solved the problem.
